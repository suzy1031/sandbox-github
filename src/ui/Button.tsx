import { useState } from 'react';

export const Button = ({ primary = true }) => {
	const [isToggleOn, setIsToggleOn] = useState(true);

	const className = primary ? 'primary' : 'secondary';

	return (
		<button
			className={className}
			type='button'
			onClick={() => setIsToggleOn(!isToggleOn)}
		>
			{isToggleOn ? 'ON' : 'OFF'}
		</button>
	);
};
