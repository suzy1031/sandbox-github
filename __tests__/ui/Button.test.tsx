import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { Button } from '../../src/ui/Button';
import { describe, it, expect } from 'vitest';

describe('button', () => {
	it('renders correctly with react-test-renderer', () => {
		const button = renderer.create(<Button />);
		expect(button.toJSON()).toMatchSnapshot();
	});

	it('change the button text upon clicking the button using React Testing Library', () => {
		const button = render(<Button />);
		fireEvent.click(button.getByText('ON'));
		expect(button.getByText(/OFF/i)).toBeTruthy();
		fireEvent.click(button.getByText('OFF'));
		expect(button.getByText(/ON/i)).toBeTruthy();
	});
});
