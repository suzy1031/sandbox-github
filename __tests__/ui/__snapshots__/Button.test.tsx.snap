// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > renders correctly with react-test-renderer 1`] = `
<button
  className="primary"
  onClick={[Function]}
  type="button"
>
  ON
</button>
`;
